tf/templates/global_variables.tmpl

```diff
@@ -26,3 +26,4 @@ dns_ruleset_name              : ${dns-ruleset-name}
 scheduler_name                : ${scheduler-name}
```

Comment:
I think this can be simplified because the VM names are not an output of the deployment process but are defined in the config.yml file. So I would remove all these from the global template file, and instead read these values from the config file. Ansible playbooks loads the config file as global variables so it should be straightforward to use this technics.
Doing it will also avoid the need to redeploy in order to rebuild the all.yml file on an existing environment.

Reply:
True, however all these values are optional, and this will add more logic to the ansible playbooks. How do we deal with the default values so that they don't appear in many places?

Reply:
Best would be to use this default syntax
"{{ ondemand.name | default('ondemand') }}"